diff --git a/dist/PrincipalFactory.js b/dist/PrincipalFactory.js
index ed6b6c85d09c8262c47632a24f72600a42758042..62c61eeacd576d60e34af5318658fb2c95f0d5f9 100644
--- a/dist/PrincipalFactory.js
+++ b/dist/PrincipalFactory.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { ProjectPrincipal } from './ProjectPrincipal.js';
 import { debugLog } from './util/debug.js';
 import { toRelative } from './util/path.js';
diff --git a/dist/ProjectPrincipal.js b/dist/ProjectPrincipal.js
index 43a8c29d0c43563e8b49962bee36619a5dd0a1c6..cd222aea57cfa8864f07c92285e930fa48f44488 100644
--- a/dist/ProjectPrincipal.js
+++ b/dist/ProjectPrincipal.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { CacheConsultant } from './CacheConsultant.js';
 import { getCompilerExtensions } from './compilers/index.js';
 import { ANONYMOUS, DEFAULT_EXTENSIONS, PUBLIC_TAG } from './constants.js';
diff --git a/dist/plugins/astro/resolveFromAST.js b/dist/plugins/astro/resolveFromAST.js
index 073ef0550f05ce9172d0aab46df17c24be248971..83e50242bd48ad8094a3d960521f476eeec28c85 100644
--- a/dist/plugins/astro/resolveFromAST.js
+++ b/dist/plugins/astro/resolveFromAST.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { getPropertyValues } from '../../typescript/ast-helpers.js';
 export const getSrcDir = (sourceFile) => {
     const srcDir = 'src';
diff --git a/dist/plugins/next/resolveFromAST.js b/dist/plugins/next/resolveFromAST.js
index f3c8295a1f7463c58fbc89ab47ee404e013e7e32..1b4dc5c1b49bcabb9e488f3846f91dae3e64a4ab 100644
--- a/dist/plugins/next/resolveFromAST.js
+++ b/dist/plugins/next/resolveFromAST.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { getPropertyValues } from '../../typescript/ast-helpers.js';
 export const getPageExtensions = (sourceFile) => {
     const pageExtensions = new Set();
diff --git a/dist/plugins/sst/resolveFromAST.js b/dist/plugins/sst/resolveFromAST.js
index 4380454f650a52ca6c4787c7a1b7cbf267af487d..0bdf6ee1cae259e2a744a039217a6141aa5463f0 100644
--- a/dist/plugins/sst/resolveFromAST.js
+++ b/dist/plugins/sst/resolveFromAST.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { getImportMap, getPropertyValues } from '../../typescript/ast-helpers.js';
 import { toDeferResolveProductionEntry } from '../../util/input.js';
 export const getInputsFromHandlers = (sourceFile, options) => {
diff --git a/dist/plugins/starlight/resolveFromAST.js b/dist/plugins/starlight/resolveFromAST.js
index 22187ed087cc20e2a7dc61a463e26fc9d052458f..b8d85f0af14cb57d79786a5ea22c2ed63ce21e4a 100644
--- a/dist/plugins/starlight/resolveFromAST.js
+++ b/dist/plugins/starlight/resolveFromAST.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { getDefaultImportName, getImportMap, getPropertyValues } from '../../typescript/ast-helpers.js';
 export const getComponentPathsFromSourceFile = (sourceFile) => {
     const componentPaths = new Set();
diff --git a/dist/plugins/vite/helpers.js b/dist/plugins/vite/helpers.js
index 5bd64a2eae3721a421b2418ea2ee1009bb515f7b..5ae2ebf62330a3ce6e61b63214335a211b71b895 100644
--- a/dist/plugins/vite/helpers.js
+++ b/dist/plugins/vite/helpers.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { findDescendants, getDefaultImportName, getImportMap, stripQuotes } from '../../typescript/ast-helpers.js';
 export const getReactBabelPlugins = (sourceFile) => {
     const babelPlugins = [];
diff --git a/dist/typescript/SourceFileManager.js b/dist/typescript/SourceFileManager.js
index 9a2969cc73bcca1a2de10a592d38a1e0a476d3fa..e3c3d2d8f5c152b5f6e5c075711d3b5a1ee20c9b 100644
--- a/dist/typescript/SourceFileManager.js
+++ b/dist/typescript/SourceFileManager.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { FOREIGN_FILE_EXTENSIONS } from '../constants.js';
 import { debugLog } from '../util/debug.js';
 import { extname, isInNodeModules, isInternal } from '../util/path.js';
diff --git a/dist/typescript/ast-helpers.js b/dist/typescript/ast-helpers.js
index bb2169cf25e88039fde524e240da68cd8a712d8d..4f6646fcc1459e86309eaa6f53fa29ee99a2ff80 100644
--- a/dist/typescript/ast-helpers.js
+++ b/dist/typescript/ast-helpers.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { FIX_FLAGS, SYMBOL_TYPE } from '../constants.js';
 function isGetOrSetAccessorDeclaration(node) {
     return node.kind === ts.SyntaxKind.SetAccessor || node.kind === ts.SyntaxKind.GetAccessor;
diff --git a/dist/typescript/create-hosts.js b/dist/typescript/create-hosts.js
index cfec15a10ab0021dbe0bdc4447742f17352b2e37..8845a818fee6435e3bba08c510f0360c48e9b03c 100644
--- a/dist/typescript/create-hosts.js
+++ b/dist/typescript/create-hosts.js
@@ -1,6 +1,6 @@
 import { EOL } from 'node:os';
 import path from 'node:path';
-import ts from 'typescript';
+import ts from 'typescript5';
 import { getCompilerExtensions } from '../compilers/index.js';
 import { createCustomModuleResolver } from './resolve-module-names.js';
 const libLocation = path.dirname(ts.getDefaultLibFilePath({}));
diff --git a/dist/typescript/find-internal-references.js b/dist/typescript/find-internal-references.js
index 31c72d977fd8561c7ebfa91781396453bc4dd4b3..f6a9dbb116db74294ba025fa6ed1cd8f088802a5 100644
--- a/dist/typescript/find-internal-references.js
+++ b/dist/typescript/find-internal-references.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { isIdChar } from '../util/regex.js';
 export const isType = (item) => item.type === 'type' || item.type === 'interface' || item.type === 'enum';
 const findInFlow = (flowNode, targetSymbol) => {
diff --git a/dist/typescript/get-imports-and-exports.js b/dist/typescript/get-imports-and-exports.js
index 9a2749f2007aa895f3b15f6fc82c2058c707ac59..23083ad2e285a9c21e9e3cc04c53c4f35a38c5c3 100644
--- a/dist/typescript/get-imports-and-exports.js
+++ b/dist/typescript/get-imports-and-exports.js
@@ -1,5 +1,5 @@
 import { isBuiltin } from 'node:module';
-import ts from 'typescript';
+import ts from 'typescript5';
 import { ALIAS_TAG, IMPORT_MODIFIERS, IMPORT_STAR, OPAQUE, PROTOCOL_VIRTUAL, SIDE_EFFECTS } from '../constants.js';
 import { addNsValue, addValue, createImports } from '../util/module-graph.js';
 import { getPackageNameFromFilePath, isStartsLikePackageName, sanitizeSpecifier } from '../util/modules.js';
diff --git a/dist/typescript/resolve-module-names.js b/dist/typescript/resolve-module-names.js
index 5bfdc1889d005647989fc8713a05453df85dc6eb..960ca930eb544ea4386b6c960b35dfc6f9f6ff26 100644
--- a/dist/typescript/resolve-module-names.js
+++ b/dist/typescript/resolve-module-names.js
@@ -1,6 +1,6 @@
 import { existsSync } from 'node:fs';
 import { isBuiltin } from 'node:module';
-import ts from 'typescript';
+import ts from 'typescript5';
 import { DEFAULT_EXTENSIONS } from '../constants.js';
 import { sanitizeSpecifier } from '../util/modules.js';
 import { timerify } from '../util/Performance.js';
diff --git a/dist/typescript/visitors/dynamic-imports/importCall.js b/dist/typescript/visitors/dynamic-imports/importCall.js
index fc38f360c0a8c77a8e51364500c8895eea681e15..9ff058477a392f9f7184f88e1ac53f04c8a154ff 100644
--- a/dist/typescript/visitors/dynamic-imports/importCall.js
+++ b/dist/typescript/visitors/dynamic-imports/importCall.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { IMPORT_MODIFIERS } from '../../../constants.js';
 import { findAncestor, findDescendants, getAccessedIdentifiers, isAccessExpression, isImportCall, isInOpaqueExpression, isTopLevel, stripQuotes, } from '../../ast-helpers.js';
 import { importVisitor as visit } from '../index.js';
diff --git a/dist/typescript/visitors/dynamic-imports/importType.js b/dist/typescript/visitors/dynamic-imports/importType.js
index 33ef88bd44d83a57df81a782a6fb520959714770..4a90262ccb2e1748189b54b88c6d898246a9eae1 100644
--- a/dist/typescript/visitors/dynamic-imports/importType.js
+++ b/dist/typescript/visitors/dynamic-imports/importType.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { IMPORT_MODIFIERS } from '../../../constants.js';
 import { importVisitor as visit } from '../index.js';
 export default visit(() => true, node => {
diff --git a/dist/typescript/visitors/dynamic-imports/jsDocType.js b/dist/typescript/visitors/dynamic-imports/jsDocType.js
index a273e452d94778676e059fc4873256bcecdddd72..15f365b8db00b447bd0f14703da89d81f005db0a 100644
--- a/dist/typescript/visitors/dynamic-imports/jsDocType.js
+++ b/dist/typescript/visitors/dynamic-imports/jsDocType.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { IMPORT_MODIFIERS, IMPORT_STAR } from '../../../constants.js';
 import { importVisitor as visit } from '../index.js';
 const supportsJSDocImportTag = 'isJSDocImportTag' in ts;
diff --git a/dist/typescript/visitors/dynamic-imports/requireCall.js b/dist/typescript/visitors/dynamic-imports/requireCall.js
index bb16a51f3ae54b531c9f05f3a557b2f9826f6d28..7c4ba1f835098029e748d97d4c371c16d7d00b36 100644
--- a/dist/typescript/visitors/dynamic-imports/requireCall.js
+++ b/dist/typescript/visitors/dynamic-imports/requireCall.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { IMPORT_MODIFIERS, IMPORT_STAR } from '../../../constants.js';
 import { findAncestor, findDescendants, isModuleExportsAccess, isRequireCall, isTopLevel } from '../../ast-helpers.js';
 import { importVisitor as visit } from '../index.js';
diff --git a/dist/typescript/visitors/dynamic-imports/resolveCall.js b/dist/typescript/visitors/dynamic-imports/resolveCall.js
index ad4f2d21c617265e09fbc9a157e8a33a5221e22c..5f0faf99d26ae4791717f8649aa5da8fbb62e445 100644
--- a/dist/typescript/visitors/dynamic-imports/resolveCall.js
+++ b/dist/typescript/visitors/dynamic-imports/resolveCall.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { IMPORT_MODIFIERS } from '../../../constants.js';
 import { isPropertyAccessCall } from '../../ast-helpers.js';
 import { importVisitor as visit } from '../index.js';
diff --git a/dist/typescript/visitors/dynamic-imports/urlConstructor.js b/dist/typescript/visitors/dynamic-imports/urlConstructor.js
index ecc6ea873c525d7987e66b3e7a796e930c8ab587..7579c0d890f7b704cfe235d4686c621290441362 100644
--- a/dist/typescript/visitors/dynamic-imports/urlConstructor.js
+++ b/dist/typescript/visitors/dynamic-imports/urlConstructor.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { IMPORT_MODIFIERS } from '../../../constants.js';
 import { importVisitor as visit } from '../index.js';
 export default visit(() => true, node => {
diff --git a/dist/typescript/visitors/exports/exportAssignment.js b/dist/typescript/visitors/exports/exportAssignment.js
index 7af549dbb734e298c8f82f71bbec2b5d1c0acefa..aa54d517833ca8e479794fff8f486342f52dee58 100644
--- a/dist/typescript/visitors/exports/exportAssignment.js
+++ b/dist/typescript/visitors/exports/exportAssignment.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { FIX_FLAGS } from '../../../constants.js';
 import { getClassMember, getEnumMember, getNodeType, isNonPrivatePropertyOrMethodDeclaration, } from '../../ast-helpers.js';
 import { isModule } from '../helpers.js';
diff --git a/dist/typescript/visitors/exports/exportDeclaration.js b/dist/typescript/visitors/exports/exportDeclaration.js
index 943db599f4d5d49870d3ece8adff5bd41de6facd..5515e6d4948ac9093d05ffa6d633c272ab56d5ea 100644
--- a/dist/typescript/visitors/exports/exportDeclaration.js
+++ b/dist/typescript/visitors/exports/exportDeclaration.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { FIX_FLAGS, SYMBOL_TYPE } from '../../../constants.js';
 import { isModule } from '../helpers.js';
 import { exportVisitor as visit } from '../index.js';
diff --git a/dist/typescript/visitors/exports/exportKeyword.js b/dist/typescript/visitors/exports/exportKeyword.js
index 458a39d0906f25c8472ab21b22846df1da059221..ece9d0f00d70a4d9dbd2282e76ec62fe9e556241 100644
--- a/dist/typescript/visitors/exports/exportKeyword.js
+++ b/dist/typescript/visitors/exports/exportKeyword.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { FIX_FLAGS, SYMBOL_TYPE } from '../../../constants.js';
 import { compact } from '../../../util/array.js';
 import { getClassMember, getDefaultKeywordNode, getEnumMember, getExportKeywordNode, isNonPrivatePropertyOrMethodDeclaration, } from '../../ast-helpers.js';
diff --git a/dist/typescript/visitors/exports/exportsAccessExpression.js b/dist/typescript/visitors/exports/exportsAccessExpression.js
index 90a7974772ca02172db68eaf84cbfb011523a155..da79d35ec453001599671c7f285a085c129c4b98 100644
--- a/dist/typescript/visitors/exports/exportsAccessExpression.js
+++ b/dist/typescript/visitors/exports/exportsAccessExpression.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { FIX_FLAGS, SYMBOL_TYPE } from '../../../constants.js';
 import { isJS } from '../helpers.js';
 import { exportVisitor as visit } from '../index.js';
diff --git a/dist/typescript/visitors/exports/moduleExportsAccessExpression.js b/dist/typescript/visitors/exports/moduleExportsAccessExpression.js
index 5107d282d06da08ec8139eef812e2c031d189252..685c03d0bafb6f828d1c81b5d6fb94c62ac84c42 100644
--- a/dist/typescript/visitors/exports/moduleExportsAccessExpression.js
+++ b/dist/typescript/visitors/exports/moduleExportsAccessExpression.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { FIX_FLAGS, SYMBOL_TYPE } from '../../../constants.js';
 import { hasRequireCall, isModuleExportsAccess, stripQuotes } from '../../ast-helpers.js';
 import { isJS } from '../helpers.js';
diff --git a/dist/typescript/visitors/helpers.js b/dist/typescript/visitors/helpers.js
index 175f4b8fafdc6f7bc8140b9c3167aa9f9bdfd8f6..5b95eaab465b777bb6efba4b487d34e3cbd8e2ae 100644
--- a/dist/typescript/visitors/helpers.js
+++ b/dist/typescript/visitors/helpers.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 export const isNotJS = (sourceFile) => !isJS(sourceFile);
 export const isJS = (sourceFile) => sourceFile.scriptKind === ts.ScriptKind.JS || sourceFile.scriptKind === ts.ScriptKind.JSX;
 export const isModule = (sourceFile) => ts.isExternalModule(sourceFile);
diff --git a/dist/typescript/visitors/imports/importDeclaration.js b/dist/typescript/visitors/imports/importDeclaration.js
index 49006d760567400c3b287ed4a5b5acb2a04aebb3..b1d4824ee7f41ac06e7fc3d63fac33627d9df500 100644
--- a/dist/typescript/visitors/imports/importDeclaration.js
+++ b/dist/typescript/visitors/imports/importDeclaration.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { IMPORT_MODIFIERS, IMPORT_STAR } from '../../../constants.js';
 import { isDefaultImport } from '../../ast-helpers.js';
 import { importVisitor as visit } from '../index.js';
diff --git a/dist/typescript/visitors/imports/importEqualsDeclaration.js b/dist/typescript/visitors/imports/importEqualsDeclaration.js
index bc1611f9f81cad60114823bc3b25540120b21b4a..dfddfab76b93da9108cb839372e90864807b24b0 100644
--- a/dist/typescript/visitors/imports/importEqualsDeclaration.js
+++ b/dist/typescript/visitors/imports/importEqualsDeclaration.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { IMPORT_MODIFIERS } from '../../../constants.js';
 import { isNotJS } from '../helpers.js';
 import { importVisitor as visit } from '../index.js';
diff --git a/dist/typescript/visitors/imports/reExportDeclaration.js b/dist/typescript/visitors/imports/reExportDeclaration.js
index ee12087df2345162c9d7eb0c57438ed873e233e5..fee7d570f06b8bef573a5f453eedb746047a0afa 100644
--- a/dist/typescript/visitors/imports/reExportDeclaration.js
+++ b/dist/typescript/visitors/imports/reExportDeclaration.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { IMPORT_MODIFIERS, IMPORT_STAR } from '../../../constants.js';
 import { importVisitor as visit } from '../index.js';
 export default visit(() => true, node => {
diff --git a/dist/typescript/visitors/scripts/bun.js b/dist/typescript/visitors/scripts/bun.js
index f4d6356ca8988d9d9f1fddb0a57e714a0bc68d89..76a985bef8e4e0fbcb7db3e3325ea13ae6435011 100644
--- a/dist/typescript/visitors/scripts/bun.js
+++ b/dist/typescript/visitors/scripts/bun.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { stripQuotes } from '../../ast-helpers.js';
 import { hasImportSpecifier } from '../helpers.js';
 import { scriptVisitor as visit } from '../index.js';
diff --git a/dist/typescript/visitors/scripts/execa.js b/dist/typescript/visitors/scripts/execa.js
index 3e70e247f691bc5fdd506bb4a38113e4f43de6d1..dabc57b62ccf8257db38199ad8ce5e4694b24b83 100644
--- a/dist/typescript/visitors/scripts/execa.js
+++ b/dist/typescript/visitors/scripts/execa.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { stripQuotes } from '../../ast-helpers.js';
 import { hasImportSpecifier } from '../helpers.js';
 import { scriptVisitor as visit } from '../index.js';
diff --git a/dist/typescript/visitors/scripts/zx.js b/dist/typescript/visitors/scripts/zx.js
index 9ae2c12cbd5f1200dd464f735ddd98a5db50a380..9308f03a82cbe524062eccc6e1695a9da5f36dd7 100644
--- a/dist/typescript/visitors/scripts/zx.js
+++ b/dist/typescript/visitors/scripts/zx.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { stripQuotes } from '../../ast-helpers.js';
 import { scriptVisitor as visit } from '../index.js';
 export default visit(sourceFile => ts.getShebang(sourceFile.text) === '#!/usr/bin/env zx', node => {
diff --git a/dist/util/load-tsconfig.js b/dist/util/load-tsconfig.js
index 74f99e539c519334af425d911735d831a0e6e5a3..b41416bcffef6428adcf35bc49614e49c94b30b7 100644
--- a/dist/util/load-tsconfig.js
+++ b/dist/util/load-tsconfig.js
@@ -1,4 +1,4 @@
-import ts from 'typescript';
+import ts from 'typescript5';
 import { isFile } from './fs.js';
 import { dirname } from './path.js';
 const dtsMatch = /\.d\.(c|m)?ts$/;
diff --git a/package.json b/package.json
index 0f6147a76a6a0a00d514ed09cdc2a13f47301309..7445b3fc6be91e18eea4a8f6644df749efcec53a 100644
--- a/package.json
+++ b/package.json
@@ -68,11 +68,11 @@
     "picomatch": "^4.0.1",
     "smol-toml": "^1.5.2",
     "strip-json-comments": "5.0.3",
+    "typescript5": "npm:typescript@5.9.2",
     "zod": "^4.1.11"
   },
   "peerDependencies": {
-    "@types/node": ">=18",
-    "typescript": ">=5.0.4 <7"
+    "@types/node": ">=18"
   },
   "devDependencies": {
     "@jest/types": "^29.6.3",
